### DISCUSSION TITLE
Word list ideas

### DISCUSSION THREAD

**Discussion by silverwings15** [sajadhsm/new-word-tab#25] [Ideas]
About time I got around to this.. Anyway, I've been doing some thinking, and now I think that having _some_ duplicate words across different lists isn't the end of the world at all, for the following reasons:

- It's not that huge an inconvenience to users, especially if they will see from which list the word comes from. They can mark it again as learned, no big deal.
- In some rare cases it might even be a word that the user forgot about, therefore it will be beneficial for them
- It's impossible to avoid duplication of words anyway, even if we had the manpower to doublecheck everything, especially if we use 'objective' lists instead of subjective ones. With subjective lists (e.g. words at a skill level, words in a certain exam etc.) we can at least modify it to avoid duplicates, but we can't do that for objective lists e.g. words from a book, specialized terms pertaining to a subject, etc.

With that in mind, I'm considering the 3 following categories for each possible word list:
1. Fun/Games. For this category I'm thinking the complete list of Wordle answers: https://gist.github.com/cfreshman/a03ef2cba789d8cf00c08f767e0fad7b, also lists of animals, kitchenware, fruits, etc.. you get the idea
2. Educational (general): For starters, I'll pull all the words from the Sadlier-Oxford Vocabulary Workshop series, which is used in American schools https://www.sadlier.com/school/vw1-12
3. Educational (specialized): Specialized terms for a profession, subject, etc. e.g. https://www.amazon.com/Dictionary-Diplomacy-G-Berridge/dp/1403915369. The problem with this is that sometimes the terms have many different meanings in ordinary contexts, and so the pertinent definitions will not be highlighted and instead buried below other more common definitions

LMK your thoughts!

**Comment by sajadhsm** [comment]
Thanks. The reasons you provided make sense.

For showing which list the word is from, as discussed before it's better to go with the first solution in https://github.com/sajadhsm/new-word-tab/issues/20#issuecomment-1047632031

Now, the way the ignored list is handled currently, can be a problem or not:

If we ignore a word from a list, then only the word itself is added to the ignored list, not its relation to the word list that it came from.
When a random word is picked again, if that word is already in the ignored list, it will be skipped and not picked from any list. So the duplicated words won't have a higher chance of being displayed if the word is marked as ignored. (The second reason you mentioned will never happen.)

But for example, I've added a custom Fun word list that contains a word that I've already marked as ignored from another list.
I will never see that word in my tabs, but maybe I want to see it if I ONLY selected my custom word list as the words poll.

I think we need the concept of **GLOBAL IGNORE** and **LOCAL IGNORE** to cover both cases. But It adds more complexity both for users and the codebase. Because the list relation needs to stored as well and also the UI should be easily recognizable.

Nevertheless, I will add the ability to add custom word lists. The categories you mentioned are great and in the future maybe we can include the lists that you are working on, directly in the source code so all the users can enjoy them :D

I think for now we can explicitly treat the ignore action as GLOBAL IGNORE and if we feel that the other scenario is needed, add the local ignore concept later. But the user needs to manually manage the words they already globally ignored.

**Comment by silverwings15** [comment]
> If we ignore a word from a list, then only the word itself is added to the ignored list, not its relation to the word list that it came from. When a random word is picked again, if that word is already in the ignored list, it will be skipped and not picked from any list. So the duplicated words won't have a higher chance of being displayed if the word is marked as ignored. (The second reason you mentioned will never happen.)

hmm true, i didn't think of this.

> The categories you mentioned are great and in the future maybe we can include the lists that you are working on, directly in the source code so all the users can enjoy them :D

yeah, the word lists i'm proposing are all meant to be included as default, hence why i wanted your opinion on it (e.g. if you're good with the Vocabulary Workshop content, i'll work to pull the word lists from there).  we don't have to create the actual categories in the addon, it's just a parameter for us to easily consider what lists we can possibly add. but if you think that's something that can be useful to implement, then by all means go ahead

> I think for now we can explicitly treat the ignore action as GLOBAL IGNORE and if we feel that the other scenario is needed, add the local ignore concept later. But the user needs to manually manage the words they already globally ignored.

sounds like a good solution 👍‍

**Reply by sajadhsm** [reply]
I'm good with any extra list. Giving the users more built-in lists will definitely add great value.

We will do a quick review on the word lists that you create and if everything is OK, they will be added to the addon. :D

**Reply by silverwings15** [reply]
sure, i'll get started on that

**Comment by sajadhsm** [comment]
For the built-in lists, we can also provide a description sentence. Kinda informs the user about what to expect from the list.

![photo_2022-05-19_21-36-34](https://user-images.githubusercontent.com/20022818/169358130-563b7bea-3afc-4283-a8a4-d2f17b88facf.jpg)

For the custom list I have such UI in mind:

![photo_2022-05-19_21-36-26](https://user-images.githubusercontent.com/20022818/169358099-9af23e7e-2d2d-43e4-a53e-080e598628ed.jpg)

After pressing the _Add List_ button, a dialog shows up and asks for the list file and its display name.
Custom lists will have a delete button too. Maybe it will be handy to add an edit button so users can add, modify or remove words to the list in the addon itself.

**Reply by silverwings15** [reply]
looks great!
